graph/path/yen_ksp.go

```diff
@@ -0,0 +1,144 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"fmt"
+	"sort"
+
+	"gonum.org/v1/gonum/graph"
+)
+
+type YenShortest struct {
+	path   []graph.Node
+	weight float64
+}
+
+func YenKSP(source graph.Node, sink graph.Node, g graph.Graph, k int) [][]graph.Node {
+	yk := yenKSPAdjuster{
+		g:       g,
+		visited: make([]int64, 0),
+		from:    g.From,
+		weight:  g.(graph.Weighted).Weight,
```

Comment:
This will panic if g is not a `graph.Weighted`. If it's the case that you always expect g to be a weighted graph, specify that in the signature.